clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2813,7 +2838,10 @@ void handleResponse(AbstractResponse abstractResponse) {
 
                 // If coordinator changed since we fetched it, retry
                 if (ConsumerGroupOperationContext.hasCoordinatorMoved(response)) {
-                    rescheduleFindCoordinatorTask(context, () -> getDescribeConsumerGroupsCall(context));
+                    long nextAllowedTryMs = calculateNextAllowedRetryMs(this.nextAllowedTryMs());
+                    Call call = getDescribeConsumerGroupsCall(context, Optional.of(this.tries() + 1),
```

Comment:
I'm just curious that can we call Call::fail to push the call instance into the queue again?

Reply:
We won't be able to do this because we need to be able to reschedule the `findCoordinator` call and then subsequently schedule the call. It's easier to pass the state across a new construction of the call and have the fail logic when we are in the `findCoordinator` logic.